tests/data/test_custom_datasets.py

```diff
@@ -0,0 +1,680 @@
+from omegaconf import DictConfig, OmegaConf
+import sleap_io as sio
+from sleap_nn.data.custom_datasets import (
+    BottomUpDataset,
+    CenteredInstanceDataset,
+    CentroidDataset,
+    SingleInstanceDataset,
+)
+
+
+def test_bottomup_dataset(minimal_instance):
+    """Test BottomUpDataset class."""
+    base_bottom_config = OmegaConf.create(
+        {
+            "user_instances_only": True,
+            "preprocessing": {
+                "max_height": None,
+                "max_width": None,
+                "scale": 1.0,
+                "is_rgb": False,
+            },
+            "use_augmentations_train": False,
+        }
+    )
+
+    confmap_head = DictConfig({"sigma": 1.5, "output_stride": 2})
+    pafs_head = DictConfig({"sigma": 4, "output_stride": 4})
+
+    dataset = BottomUpDataset(
+        data_config=base_bottom_config,
+        max_stride=32,
+        confmap_head_config=confmap_head,
+        pafs_head_config=pafs_head,
+        labels=sio.load_slp(minimal_instance),
+        apply_aug=base_bottom_config.use_augmentations_train,
+    )
+
+    gt_sample_keys = [
+        "image",
+        "instances",
+        "video_idx",
+        "frame_idx",
+        "confidence_maps",
+        "orig_size",
+        "num_instances",
+        "part_affinity_fields",
+    ]
+    sample = next(iter(dataset))
+    assert len(sample.keys()) == len(gt_sample_keys)
+
+    for gt_key, key in zip(sorted(gt_sample_keys), sorted(sample.keys())):
+        assert gt_key == key
+    assert sample["image"].shape == (1, 1, 384, 384)
+    assert sample["confidence_maps"].shape == (1, 2, 192, 192)
+    assert sample["part_affinity_fields"].shape == (96, 96, 2)

```

Comment:
Shouldn't this be (2, 96, 96) since pytorch is channels first?

Reply:
This is fixed in Pr: #121